### PR TITLE
PYTHON-4804 Migrate test_comment.py to async

### PR DESCRIPTION
**Pull request by sleepyStick**

### CODE REVIEW

**Comment by sleepyStick** [comment]
Not sure why the link check is failing...I dont think its anything I did?

**Comment by blink1073** [comment]
> Not sure why the link check is failing...I don't think its anything I did?

No, the docs team is working on a fix

File: test/asynchronous/test_comment.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                        await destruct_coll.drop()
+                    elif h == db.validate_collection:
+                        coll = db.get_collection("test")
+                        await coll.insert_one({})
+                        maybe_cursor = await db.validate_collection(*args, **kwargs)
+                    else:
+                        if iscoroutinefunction(coll.create_index):
+                            await coll.create_index("a")
+                        else:
+                            coll.create_index("a")
```
Comment (NoahStapp) [inline test/asynchronous/test_comment.py:74]:
```suggestion
                        if not _IS_SYNC and isinstance(coll, Empty):
                            coll.create_index("a")
                        else:
                            await coll.create_index("a")
```
For clarity so unfamiliar readers can understand why `coll.create_index` could be synchronous.

File: test/asynchronous/test_comment.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                        if iscoroutinefunction(h):
+                            maybe_cursor = await h(*args, **kwargs)
+                        else:
+                            maybe_cursor = h(*args, **kwargs)
```
Comment (NoahStapp) [inline test/asynchronous/test_comment.py:78]:
```suggestion
                        if not _IS_SYNC and iscoroutinefunction(h):
                            maybe_cursor = await h(*args, **kwargs)
                        else:
                            maybe_cursor = h(*args, **kwargs)
```
Same here.

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: test/asynchronous/test_comment.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        coll=Empty(),  # noqa: B008
+    ):
+        for h, args in helpers:
+            c = "testing comment with " + h.__name__
+            with self.subTest("collection-" + h.__name__ + "-comment"):
+                for cc in [c, {"key": c}, ["any", 1]]:
+                    listener.reset()
+                    kwargs = {"comment": cc}
+                    if h == coll.rename:
+                        _ = await db.get_collection("temp_temp_temp").drop()
```
[the remark this answers is not in the document]
Comment (sleepyStick) [reply test/asynchronous/test_comment.py]: probably not! just removed it :)

**Review by NoahStapp — APPROVED** [review]
LGTM!

**Comment by sleepyStick** [comment]
Oops, didn't realize that merging would dismiss the review...sorry!

File: test/test_comment.py [hunk 4]
```diff
@@ -20,14 +20,16 @@
...
 from test.utils import EventListener
 
 from bson.dbref import DBRef
 from pymongo.operations import IndexModel
 from pymongo.synchronous.command_cursor import CommandCursor
 
+_IS_SYNC = True
+
 
 class Empty:
```
Comment (ShaneHarvey) [inline test/test_comment.py:34]: This Empty class looks very suspicious. What's the point of it?

File: test/test_comment.py [hunk 5]
```diff
@@ -66,8 +68,14 @@ def _test_ops(
                         coll.insert_one({})
                         maybe_cursor = db.validate_collection(*args, **kwargs)
                     else:
-                        coll.create_index("a")
-                        maybe_cursor = h(*args, **kwargs)
+                        if not _IS_SYNC and isinstance(coll, Empty):
+                            coll.create_index("a")
```
Comment (ShaneHarvey) [inline test/test_comment.py:72]: How can we call create_index when coll is `Empty`?

File: test/test_comment.py (hunk 4 above)
Comment (sleepyStick) [reply test/test_comment.py:34]: Uhhh honestly not sure. It was added in [PYTHON-2682](https://jira.mongodb.org/browse/PYTHON-2682).

Comment (sleepyStick) [reply test/test_comment.py]: cleaned up `_test_ops` and now `Empty` isn't needed anymore :)

File: test/test_comment.py (hunk 5 above)
Comment (sleepyStick) [reply test/test_comment.py]: Thats a great question...HAHA code is removed now.

**Review by ShaneHarvey — DISMISSED** [review]
LGTM assuming the tests pass.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**